main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/config"
+	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/service"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+func main() {
+	log.Namespace = "dp-frontend-filter-flex-dataset"
+	ctx := context.Background()
+
+	if err := run(ctx); err != nil {
+		log.Fatal(ctx, "application unexpectedly failed", err)
+		os.Exit(1)
+	}
+
+	os.Exit(0)
+}
+
+func run(ctx context.Context) error {
+	// Create error channel for os signals
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+
+	// Create service initialiser and an error channel for service errors
+	svcList := service.NewServiceList(&service.Init{})
+	svcErrors := make(chan error, 1)
+
+	// Read config
+	cfg, err := config.Get()
+	if err != nil {
+		log.Error(ctx, "unable to retrieve service configuration", err)
```

Comment:
Logging errors and then also returning the same error? Please kill it with fire! We don't need errors being logged random amounts of times. Errors should be returned all the way up the call stack and logged once.
If you want to add some descriptive context (and you do) you should combine these 2 statements like this:
```
return fmt.Errorf("unable to retrieve service configuration: %w", err)
```
This returns the original error wrapped in the extra message and is a far preferable way of doing things.